HW05.Rmd

```diff
@@ -0,0 +1,133 @@
+---
+title: "HW05"
+output: github_document
+author: Bei Wang
+---
+
+```{r load pkgs, include=FALSE}
+knitr::opts_chunk$set(echo = TRUE)
+library("tidyverse")
+library("magrittr")
+library("knitr")
+library("readr")
+```
+
+## 1. Tidying the dad_mom file
+
+    Each variable must have its own column.
+    Each observation must have its own row.
+    Each value must have its own cell.
+
+```{r dad_mom}
+#read dataset
+dad_mom<- read.table("dad_mom.txt", header=TRUE) 
+
+#reshape
+tidy_dad_mom<- dad_mom %>%
+  unite("dad", name_dad, income_dad) %>%
+  unite("mom", name_mom, income_mom) %>%
+  pivot_longer(c(dad, mom), names_to="parent", values_to="income") %>%
+  separate(income, c("name", "income"))
```

Comment:
I like the way you united multiple columns into one and then transposed the data set from wide to long. I also learnt how to use pivot_longer( )function from you code!